pageserver/src/tenant/remote_timeline_client/download.rs

```diff
@@ -207,17 +207,29 @@ async fn download_object<'a>(
                         size_tracking,
                         BytesMut::with_capacity(super::BUFFER_SIZE),
                     );
-                    while let Some(res) =
-                        futures::StreamExt::next(&mut download.download_stream).await
-                    {
-                        let chunk = match res {
-                            Ok(chunk) => chunk,
-                            Err(e) => return Err(e),
-                        };
-                        buffered
-                            .write_buffered(tokio_epoll_uring::BoundedBuf::slice_full(chunk), ctx)
-                            .await?;
+
+
+                    // Cancellation: if our cancellation token fires here, we will return an error and leave
+                    // an inccomplete download on disk.  Callers are responsible for cleaning that up: for example
+                    // download_layer_file calls this function with a temporary path, so it is safe to leak that
+                    // temp file on shutdown, it is cleaned up on next startup.
+                    loop {
+                        tokio::select! {
+                            _ = cancel.cancelled() => {
+                                return Err(DownloadError::Cancelled)
+                            },
```

Comment:
The stream should handle this already.

Reply:
there is no reproducer here. I wonder what prompted this.

Reply:
looking at the issue maybe we should instead lower the current 1s limit to 100ms, hopefully catch more of slow gate closings.